backward.hpp

```diff
@@ -2431,17 +2432,25 @@ class TraceResolverLinuxImpl<trace_resolver_tag::libdwarf>
       ELF_GET_DATA(32)
     } else if (e_ident[EI_CLASS] == ELFCLASS64) {
       // libelf might have been built without 64 bit support
-#if __LIBELF64
       ELF_GET_DATA(64)
-#endif
     }
 
     if (!debuglink.empty()) {
       // We have a debuglink section! Open an elf instance on that
       // file instead. If we can't open the file, then return
       // the elf handle we had already opened.
       dwarf_file_t debuglink_file;
-      debuglink_file.reset(open(debuglink.c_str(), O_RDONLY));
+      {
+        char* object_filename_copy = strdup(object_filename.c_str());
+        const std::string directory_path = std::string(dirname(object_filename_copy));
+        free(object_filename_copy);
+        debuglink_file.reset(
+            open((directory_path + '/' + debuglink).c_str(), O_RDONLY));
```

Comment:
I expect `debuglink` to have only filename since, [according to docs](https://sourceware.org/gdb/onlinedocs/gdb/Separate-Debug-Files.html#:~:text=The%20executable%20contains%20a%20debug%20link%20that%20specifies%20the%20name%20of%20the%20separate%20debug%20info%20file.):

> The executable contains a debug link that specifies the name of the separate debug info file.

Also, `objcopy` tool writes only the filename component in the `.gnu_debuglink` section even if an absolute path is provided:
https://github.com/bminor/binutils-gdb/blob/b4edb38e827974ecdbc4a6712d8b21bf876018a2/bfd/opncls.c#L1707-L1708